src/module/item/shield/document.ts

```diff
@@ -0,0 +1,335 @@
+    }
+
+    get isBuckler(): boolean {
+        return ["buckler", "casters-targe", "dart-shield", "gauntlet-buckler", "heavy-rondache", "klar"].includes(
+            this.system.baseItem ?? "",
+        );
+    }
+
+    get isTowerShield(): boolean {
+        return ["fortress-shield", "tower-shield"].includes(this.system.baseItem ?? "");
+    }
+
+    get speedPenalty(): number {
+        return this.system.speedPenalty || 0;
+    }
+
+    get acBonus(): number {
+        return this.system.acBonus;
+    }
+
+    override get isSpecific(): boolean {
+        return !!this.system.specific;
+    }
+
+    /** Given this is a shield, is it raised? */
+    get isRaised(): boolean {
+        return (
+            !!this.actor?.isOfType("character", "npc") &&
+            this.id === this.actor.attributes.shield.itemId &&
+            this.actor.attributes.shield.raised
+        );
+    }
+
+    /** Generate a list of strings for use in predication */
+    override getRollOptions(prefix = "armor"): string[] {
+        const reinforcingRune = this.system.runes.reinforcing;
+        const reinforcingSlug = [null, "minor", "lesser", "moderate", "greater", "major", "supreme"][reinforcingRune];
+        const reinforcingOptions = {
+            [`rune:reinforcing:${reinforcingRune}`]: !!reinforcingRune,
+            [`rune:reinforcing:${reinforcingSlug}`]: !!reinforcingRune,
+        };
+
+        return [
+            super.getRollOptions(prefix),
+            Object.entries({
+                [`base:${this.baseType}`]: !!this.baseType,
+                ...reinforcingOptions,
+            })
+                .filter(([, isTrue]) => !!isTrue)
+                .map(([key]) => `${prefix}:${key}`),
+        ]
+            .flat()
+            .sort();
+    }
+
+    override prepareBaseData(): void {
+        super.prepareBaseData();
+
+        // Temporary measure during migration
+        if ("category" in this.system) {
+            this.type = "shield";
+            this.system.runes = { reinforcing: 0 };
+        }
+
+        const { material } = this;
+        const reinforcingRuneData = RUNE_DATA.shield.reinforcing;
+        const reinforcingRune = this.system.runes.reinforcing;
+        const materialData = this.isBuckler ? BUCKLER_MATERIAL_VALUATION_DATA : SHIELD_MATERIAL_VALUATION_DATA;
+
+        const adjustFromMaterialAndRune = (property: "hardness" | "maxHP", base: number): number => {
+            const fromMaterial =
+                material.type && material.grade
+                    ? materialData[material.type]?.[material.grade]?.[property] ?? base
+                    : base;
+            const additionalFromRune = reinforcingRune ? reinforcingRuneData[reinforcingRune]?.[property] : null;
+            const sumFromRune = fromMaterial + (additionalFromRune?.increase ?? 0);
+            return additionalFromRune && sumFromRune > additionalFromRune.max
+                ? Math.max(fromMaterial, additionalFromRune.max)
```

Comment:
Should this not be min? Also, I think you might be able to skip the ternary with return `Math.min(fromMaterial, additionalFromRune?.max ?? Infinity)`

Reply:
`Math.max` is correct there: a reinforcing rune limits how much it can increase a shield's hardness and HP

Reply:
But if the max is 20, and materials increase to 30, and you do Math.max(30, 20), the result is 30.

Reply:
And that would be correct